definitions/mw.lua

```diff
@@ -0,0 +1,653 @@
+-- luacheck: ignore
+---@meta mw
+mw = {}
+
+---Adds a warning which is displayed above the preview when previewing an edit. `text` is parsed as wikitext.
+---@param text string
+function mw.addWarning(text) end
+
+---Calls tostring() on all arguments, then concatenates them with tabs as separators.
+---@param ... any
+---@return string
+---@nodiscard
+function mw.allToString(...) end
+
+---Creates a deep copy of a value. All tables (and their metatables) are reconstructed from scratch. Functions are still shared, however.
+---@generic T
+---@param value T
+---@return T
+---@nodiscard
+function mw.clone(value) end
+
+---Returns the current frame object, typically the frame object from the most recent #invoke. 
+---@return Frame
+---@nodiscard
+function mw.getCurrentFrame() end
+
+---Adds one to the "expensive parser function" count, and throws an exception if it exceeds the limit (see $wgExpensiveParserFunctionLimit).
+function mw.incrementExpensiveFunctionCount() end
+
+---Returns true if the current #invoke is being substed, false otherwise. See Returning text above for discussion on differences when substing versus not substing.
```

Comment:
`See Returning text above for discussion on differences when substing versus not substing.` makes not super much sense without the link